model/lifecycle.go

```diff
@@ -658,30 +659,69 @@ func AddTasksToBuild(ctx context.Context, b *build.Build, project *Project, v *V
 	}
 
 	// update the build to hold the new tasks
-	if err := RefreshTasksCache(b.Id); err != nil {
+	if err = RefreshTasksCache(b.Id); err != nil {
 		return nil, nil, errors.Wrapf(err, "error updating task cache for '%s'", b.Id)
 	}
 
+	batchTimeTaskStatuses := []BatchTimeTaskStatus{}
+	var pRef *ProjectRef
+	if len(tasksWithBatchTime) > 0 {
+		pRef, err = FindOneProjectRef(project.Identifier)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "unable to find project ref")
+		}
+		if pRef == nil {
+			return nil, nil, errors.Errorf("project '%s' not found", project.Identifier)
+		}
+	}
+	for _, t := range tasks {
+		if !utility.StringSliceContains(tasksWithBatchTime, t.DisplayName) {
+			continue
+		}
+		activateTaskAt, err := pRef.GetActivationTimeForTask(project.FindTaskForVariant(t.DisplayName, b.BuildVariant))
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "unable to get activation time for task",
+			"variant": b.BuildVariant,
+			"task":    t.DisplayName,
+			"version": v.Id,
+		}))
+		batchTimeTaskStatuses = append(batchTimeTaskStatuses, BatchTimeTaskStatus{
+			TaskName: t.DisplayName,
+			TaskId:   t.Id,
+			ActivationStatus: ActivationStatus{
+				ActivateAt: activateTaskAt,
```

Comment:
What would `activateTaskAt` be if `pRef.GetActivationTimeForTask` errored?

Reply:
It's just time.Now (this is the same logic as the existing GetActivationTimeForVariant)